### PR TITLE
Facilitate incremental download of packages

### PR DESCRIPTION
**Pull request by mdrohmann**
https://www.pivotaltracker.com/story/show/177290892

### CODE REVIEW

File: pkg/platform/runtime/setup/implementations/alternative/runtime.go [hunk 1]
```diff
@@ -25,7 +25,9 @@ func NewSetup(store *store.Store, artifacts artifact.ArtifactRecipeMap) *Setup {
 	return &Setup{store: store, artifacts: artifacts}
 }
 
-func (s *Setup) DeleteOutdatedArtifacts(changeset artifact.ArtifactChangeset, storedArtifacted store.StoredArtifactMap) error {
+func (s *Setup) DeleteOutdatedArtifacts(changeset artifact.ArtifactChangeset, storedArtifacted store.StoredArtifactMap) ([]artifact.ArtifactID, error) {
+	var alreadyInstalled []artifact.ArtifactID
```
Comment (Naatan) [inline pkg/platform/runtime/setup/implementations/alternative/runtime.go:29]: Not going to request changes on this now, but this seems like something we should fix in the future. It's awkward that a function responsible for deleting outdated artifacts is returning artifacts that you want to keep. Either this function needs to be renamed or its logic needs to be split up into separate functions.

**Review by Naatan — APPROVED** [review]
Looks ok to me. Did you consider how this affects the env merging at the end?

File: pkg/platform/runtime/setup/implementations/alternative/runtime.go (hunk 1 above)
Comment (mdrohmann) [reply pkg/platform/runtime/setup/implementations/alternative/runtime.go]: I agree.  I pushed a new version, that also gets rid of the go-funk dependency...

**Review by Naatan — APPROVED** [review]
Much better!

**Pull request merged**